main/acle.md

```diff
@@ -4955,6 +5111,2370 @@ assertions, which may speed up access to aligned data (and will fault
+
+    *   E1 has type GNUT or cv-qualified GNUT and E2 has type VLST or
+        cv-qualified VLST.
+
+    However, the implementation does not need to diagnose these cases.
+
+    This rule avoids any ambiguity about whether the result of the
+    expression is a GNU or SVE type. For example:
+
+    ``` c
+      #if __ARM_FEATURE_SVE_BITS==512 && __ARM_FEATURE_SVE_VECTOR_OPERATORS
+      typedef int32_t vec1 __attribute__((vector_size(64)));
+      typedef svint32_t vec2 __attribute__((arm_sve_vector_bits(512)));
+      auto f(vec1 x, vec2 y) { return x + y; }
+      #endif
+    ```
+
+    is ill-formed because it is not clear whether the type of the result
+    should come from `x` or `y`. The choice would affect the ABI of
+    the function.
+
+##### `arm_sve_vector_bits` behavior specific to predicates
+
+Let VLST be a valid type:
+
+``` c
+  svbool_t __attribute__((arm_sve_vector_bits(N)))
+```
+
+VLST is then a sized type that has the following properties:
+
+``` c
+  sizeof(VLST) == N/64
+  alignof(VLST) == 2
+```
+
+If in addition the implementation defines the preprocessor macro
+
+``` c
+  __ARM_FEATURE_SVE_PREDICATE_OPERATORS
+```
+
+then:
+
+*   The GNU `__attribute__((vector_size))` extension is available.
+
+*   VLST supports the same forms of elementwise initialization as the
+    vector type:
+
+    ``` c
+      uint8_t __attribute__((vector_size(N/8)))
+    ```
+
+    except that the elements have type `bool` instead of `uint8_t`.
+
+*   VLST supports the following operators, all of which take arguments of
+    type VLST and return either a VLST or a reference to a VLST:
+
+    *   unary `~`
+
+    *   binary `&`, `|` and `^`
+
+    *   assignments `&=`, `|=` and `^=`
+
+    *   comparisons `<`, `<=`, `==`, `!=`, `>=` and `>`
+
+    All operators behave analogously to GNU vectors. For example:
+
+    ``` c
+      #if __ARM_FEATURE_SVE_BITS==256 && __ARM_FEATURE_SVE_PREDICATE_OPERATORS
+      typedef svbool_t pred __attribute__((arm_sve_vector_bits(256)));
+      auto f(pred x, pred y) { return x & y; }    // Returns a pred
+      #endif
+    ```
+
+*   VLST supports the array subscript operator `[]` in a way that
+    behaves analogously to GNU vectors, except that it is not possible to
+    take the address of the result. For example:
+
+    ``` c
+      #if __ARM_FEATURE_SVE_BITS==256 && __ARM_FEATURE_SVE_PREDICATE_OPERATORS
+      typedef svbool_t pred __attribute__((arm_sve_vector_bits(256)));
+      pred x;
+      bool f() { x[1] = true; return x[0]; }
+      #endif
+    ```
+
+*   Within a byte, subscripts count from the least significant bit. For
+    example:
+
+    ``` c
+      #if __ARM_FEATURE_SVE_BITS==256 && __ARM_FEATURE_SVE_PREDICATE_OPERATORS
+      svbool_t p __attribute__((arm_sve_vector_bits(256))) = { 0, 1 };
+      #endif
+    ```
+
+    sets the first byte of `p` to 2 for both big- and little-endian targets.
+
+An implementation may provide other operators too, but it does not need
+to do so.
```

Comment:
```suggestion
An implementation might provide other operators too, but it does not need
to do so.
```

Reply:
The meaning was more that an implementation is allowed to provide other operations too, so I changed it to “can”.